pkg/apis/ndm/v1alpha1/types.go

```diff
@@ -71,3 +71,37 @@ type StoragePoolList struct {
 
 	Items []StoragePool `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=diskobject
+
+// DiskObject describes disk resource.
+type DiskObject struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata, omitempty"`
+
+	Spec DiskObjectSpec `json:"spec"`
+}
+
+// DiskObjectSpec is the specification for the disk stored as CRD
+type DiskObjectSpec struct {
+	Uuid       string `json:"uuid"`  //unique identifier for the disk
+	Name       string `json:"name"`  //disk name (e.g. sdb, sdc)
+	Path       string `json:"path"`  //disk path (e.g. /dev/sdb)
```

Comment:
This should be an array -- as there is disk/by-id disk/by-guid (hey?! our guid=)) etc.